backend/src/main/java/capstone/kpiboard/security/SecurityConfig.java

```diff
@@ -0,0 +1,78 @@
+package capstone.kpiboard.security;
+
+import capstone.kpiboard.model.roles.TechnicalRole;
+import capstone.kpiboard.service.user.AppUserService;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.provisioning.UserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+@EnableWebSecurity
+public class SecurityConfig {
+    private final AppUserService appUserService;
+
+    public SecurityConfig(AppUserService appUserService) {
+        this.appUserService = appUserService;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        return http
+                .csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                .and()
+                .authorizeRequests()
+                .antMatchers("/api/user/login").permitAll()
+                .antMatchers("/api/user/logout").permitAll()
+                .antMatchers("/api/user/me").permitAll()
+                .antMatchers(HttpMethod.GET, "/api/roles").authenticated()
+                .antMatchers(HttpMethod.POST, "/api/roles").hasAuthority(TechnicalRole.ADMIN.toString())
+                .antMatchers(HttpMethod.DELETE, "/api/roles/*").hasAuthority(TechnicalRole.ADMIN.toString())
+                .antMatchers(HttpMethod.POST, "/api/user").hasAuthority(TechnicalRole.ADMIN.toString())
+                .antMatchers(HttpMethod.DELETE, "/api/kpis/*").hasAuthority(TechnicalRole.ADMIN.toString())
+                .antMatchers(HttpMethod.POST, "/api/kpis").hasAuthority(TechnicalRole.ADMIN.toString())
+                .antMatchers("/api/**").authenticated()
+                .anyRequest().denyAll()
+                .and().httpBasic()
+                .and().build();
+    }
+
+    @Bean
+    public UserDetailsManager userDetailsService() {
+        return new UserDetailsManager() {
```

Comment:
Ich bin mir nicht sicher, ob man diesen UserDetailsManager braucht. ICh glaub ich habe ihn gelöscht :)

Reply:
@Bean
    public AuthenticationManager authenticationManager(HttpSecurity security) throws Exception {
        return security.getSharedObject(AuthenticationManagerBuilder.class)
                .userDetailsService(appUserService)
                .passwordEncoder(encoder())
                .and().build();
    } hab ich stattdessen, glaube ich, vermutlich ist das aber egal